gnovm/pkg/gnolang/type_check.go

```diff
@@ -1142,3 +1144,164 @@ func isBlankIdentifier(x Expr) bool {
 	}
 	return false
 }
+
+// Check for invalid arguments in builtin functions
+// len, cap, make
+func AssertValidBuiltinArgType(store Store, last BlockNode, currExpr Expr) {
+	switch currExpr := currExpr.(type) {
+	case *ConstExpr:
+		AssertValidBuiltinArgType(store, last, currExpr.Source)
+	case *CallExpr:
+		ift := evalStaticTypeOf(store, last, currExpr.Func)
+		switch baseOf(ift).(type) {
+		case *FuncType:
+			validateMakeArg(store, last, currExpr)
+		}
+	}
+}
```

Comment:
`currExpr` here can only be a `callExpr`, so I guess what we do need is to check against the args in make(xxx), right?